subprojects/core/src/main/java/org/gradle/api/internal/classpath/DefaultModuleRegistry.java

```diff
@@ -161,6 +162,17 @@ private Module loadOptionalModule(String moduleName) {
 
     private Module module(String moduleName, Properties properties, Set<File> implementationClasspath) {
         String[] runtimeJarNames = split(properties.getProperty("runtime"));
+        if (JavaVersion.current().compareTo(JavaVersion.VERSION_1_9) < 0) {
```

Comment:
Do we always build the classpath for the version that is currently executing this code?

Reply:
No, not always. More specifically this won't work for worker processes, as their classpath is built by the daemon before starting the worker. Of course, our workers don't depend on JAXB, so this is not an issue. But it might confuse the reader and they might ask the same question you did.

Since we are only trying to solve this for one specific case - the AWS plugin running inside the daemon, a better place for the fix might be in the [plugin classpath provider](https://github.com/gradle/gradle/blob/82fc49648ffdcb64e0b5543c5b764037a8851f90/subprojects/core/src/main/java/org/gradle/api/internal/DynamicModulesClassPathProvider.java#L36).